core/src/avm2/verify.rs

```diff
@@ -408,12 +410,80 @@ pub fn verify_method<'gc>(
             return Err(make_error_1054(activation));
         }
 
+        let target_class = if exception.type_name.0 == 0 {
+            None
+        } else {
+            let pooled_type_name = method
+                .translation_unit()
+                .pool_maybe_uninitialized_multiname(exception.type_name, &mut activation.context)?;
+
+            if pooled_type_name.has_lazy_component() {
+                // This matches FP's error message
+                return Err(Error::AvmError(verify_error(
+                    activation,
+                    "Error #1014: Class [] could not be found.",
+                    1014,
+                )?));
+            }
+
+            let resolved_type = activation
+                .domain()
+                .get_class(&pooled_type_name, activation.context.gc_context)
+                .ok_or_else(|| {
+                    Error::AvmError(
+                        verify_error(
+                            activation,
+                            &format!(
+                                "Error #1014: Class {} could not be found.",
+                                pooled_type_name.to_qualified_name(activation.context.gc_context)
+                            ),
+                            1014,
+                        )
+                        .expect("Error should construct"),
+                    )
+                })?;
+
+            Some(resolved_type)
+        };
+
+        let variable_name = if exception.variable_name.0 == 0 {
+            None
+        } else {
+            let pooled_variable_name = method
+                .translation_unit()
+                .pool_maybe_uninitialized_multiname(
+                    exception.variable_name,
+                    &mut activation.context,
+                )?;
+
+            // FIXME: avmplus also seems to check the namespace(s)?
+            if pooled_variable_name.has_lazy_component()
+                || pooled_variable_name.is_attribute()
+                || pooled_variable_name.is_any_name()
+            {
+                // This matches FP's error message
+                return Err(make_error_1107(activation));
+            }
+
+            let namespaces = pooled_variable_name.namespace_set();
+
+            if namespaces.is_empty() {
+                // NOTE: avmplus segfaults here
+                panic!("Should have at least one namespace for QName in exception variable name");
+            }
+
+            let name = pooled_variable_name.local_name().expect("Just checked");
+
+            // avmplus uses the first namespace, regardless of how many namespaces there are.
+            Some(QName::new(namespaces[0], name))
```

Comment:
If this is a more generic rule (applying to all traits' names), we might wanna have a more generic function that converts a generic Multiname to a specific trait name.

Reply:
I have _no idea_ where else this applies, and this is short, so I think this is fine for now. If it comes up later we can always change it.